Rules.py

```diff
@@ -878,6 +878,8 @@ def no_glitches_rules(world, player):
         set_rule(world.get_entrance('Dark Lake Hylia Drop (East)', player), lambda state: state.has('Flippers', player))
         set_rule(world.get_entrance('Dark Lake Hylia Teleporter', player), lambda state: state.has('Flippers', player) and (state.has('Hammer', player) or state.can_lift_rocks(player)))
         set_rule(world.get_entrance('Dark Lake Hylia Ledge Drop', player), lambda state: state.has('Flippers', player))
+        set_rule(world.get_entrance('East Dark World Pier', player), lambda state: state.has('Flippers', player))
+        set_rule(world.get_entrance('East Dark World River Pier', player), lambda state: state.has('Flippers', player))
 
```

Comment:
This overrides the Pearl requirement from global rules. Use add_rule instead. (This is a general problem with the no_glitches_section though)

Reply:
I think this is okay because the first branch uses add_rule for East Dark World River Pier and should only be run when global_rules is called. The highlighted branch is run when the world mode is inverted and global_rules isn't called. Correct me if I'm wrong though.